src/Symfony/Component/Security/Guard/Firewall/GuardAuthenticationListener.php

```diff
@@ -121,6 +132,14 @@ private function executeGuardAuthenticator($uniqueGuardKey, GuardAuthenticatorIn
         } catch (AuthenticationException $e) {
             // oh no! Authentication failed!
 
+            if (false === $handleAuthenticationFailure) {
+                if (null !== $this->logger) {
+                    $this->logger->info('Guard authenticator {authenticator} failed to process the authentication, try the next one.', array('exception' => $e, 'authenticator' => get_class($guardAuthenticator)));
+                }
+
+                return;
+            }
+
             if (null !== $this->logger) {
```

Comment:
And perhaps add a log before `$event->setResponse()` below as well, to (eventually) inform which authenticator provided the failure handling.